### PR TITLE
Refactor Land Rights window to class

### LINKED ISSUE #13777: Refactor window to class: LandRights
See #13737 for more details.

### PR DESCRIPTION
**Pull request by GalBr**
Closes #13777

I made sure the window functions the same.
I couldn't find a park that had the construction rights button enabled, and I'm not sure how to enable it, so I couldn't test that.

### CODE REVIEW

File: src/openrct2-ui/windows/LandRights.cpp [hunk 1]
```diff
@@ -83,371 +54,354 @@ constexpr uint8_t LAND_RIGHTS_MODE_BUY_LAND = 1;
 static uint8_t _landRightsMode;
 static money32 _landRightsCost;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/LandRights.cpp:55]: These should become member variables

Comment (GalBr) [reply src/openrct2-ui/windows/LandRights.cpp]: Fixed!

**Review by tupaschoal — COMMENTED** [review]
@GalBr please make sure you mark conversations as resolved when you deal with them :)

**Comment by GalBr** [comment]
> @GalBr please make sure you mark conversations as resolved when you deal with them :)

Done, sorry! I thought that whoever commented should mark as resolved if the change looks good to them.

**Comment by tupaschoal** [comment]
If it's a clear suggestion and you implement it, then it's alright to mark as solved. If it looks like a discussion then your approach makes sense :)

File: src/openrct2-ui/windows/LandRights.cpp [hunk 2]
```diff
@@ -45,409 +46,361 @@ static rct_widget window_land_rights_widgets[] = {
...
+            return;
 
-static void WindowLandRightsUpdate(rct_window* w)
-{
-    w->frame_no++;
-    // Close window if another tool is open
-    if (!LandRightsToolIsActive())
-        window_close(w);
-}
+        int32_t size;
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/LandRights.cpp:154]: no need to declare this here. Move it to its first use on L159

File: src/openrct2-ui/windows/LandRights.cpp [hunk 3]
```diff
@@ -45,409 +46,361 @@ static rct_widget window_land_rights_widgets[] = {
...
-        window_land_rights_widgets[WIDX_BUY_CONSTRUCTION_RIGHTS].tooltip = STR_BUY_CONSTRUCTION_RIGHTS_TIP;
-    }
-}
+        SetWidgetPressed(WIDX_PREVIEW, true);
+        if (_landRightsMode == LAND_RIGHTS_MODE_BUY_LAND)
+        {
+            SetWidgetPressed(WIDX_BUY_LAND_RIGHTS, true);
+            SetWidgetPressed(WIDX_BUY_CONSTRUCTION_RIGHTS, false);
+        }
+        if (_landRightsMode == LAND_RIGHTS_MODE_BUY_CONSTRUCTION_RIGHTS)
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/LandRights.cpp:183]: `else if`

File: src/openrct2-ui/windows/LandRights.cpp [hunk 4]
```diff
@@ -45,409 +46,361 @@ static rct_widget window_land_rights_widgets[] = {
...
+            }
         }
     }
-}
 
-static bool LandRightsToolIsActive()
+private:
+    uint8_t _landRightsMode;
+    money32 _landRightsCost;
+    void InputSize()
```
Comment (duncanspumpkin) [inline src/openrct2-ui/windows/LandRights.cpp:385]: add a newline to split the memeber vars from the functions.

**Review by duncanspumpkin — CHANGES_REQUESTED** [review]
Address the minor comments and this is good to go.

File: src/openrct2-ui/windows/LandRights.cpp (hunk 2 above)
Comment (GalBr) [reply src/openrct2-ui/windows/LandRights.cpp]: Fixed.

File: src/openrct2-ui/windows/LandRights.cpp (hunk 4 above)
Comment (GalBr) [reply src/openrct2-ui/windows/LandRights.cpp:385]: Fixed.

File: src/openrct2-ui/windows/LandRights.cpp (hunk 3 above)
Comment (GalBr) [reply src/openrct2-ui/windows/LandRights.cpp]: Fixed.

**Review by duncanspumpkin — APPROVED** [review]

**Pull request merged**